yewprint/src/progressbar.rs

```diff
@@ -0,0 +1,55 @@
+use crate::Intent;
+use yew::prelude::*;
+
+pub struct ProgressBar {
+    props: Props,
+}
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct Props {
+    #[prop_or(false)]
+    pub animate: bool,
+    #[prop_or(false)]
+    pub stripes: bool,
+    #[prop_or_default]
+    pub value: Option<f32>,
+    #[prop_or_default]
+    pub intent: Option<Intent>,
+}
+
+impl Component for ProgressBar {
+    type Message = ();
+    type Properties = Props;
+
+    fn create(props: Self::Properties, _link: ComponentLink<Self>) -> Self {
+        Self { props }
+    }
+
+    fn update(&mut self, _msg: Self::Message) -> ShouldRender {
+        true
+    }
+
+    fn change(&mut self, props: Self::Properties) -> ShouldRender {
+        if self.props != props {
+            self.props = props;
+            true
+        } else {
+            false
+        }
+    }
+
+    fn view(&self) -> Html {
+        let width = if let Some(value) = self.props.value {
+            //let percent = ((1000. * value).ceil() / 10.).clamp(0.,100.);
+            let percent = ((1000. * value).ceil() / 10.).max(0.).min(100.);
+            format!("width: {}%;", percent)
+        } else {
+            "".into()
+        };
+        html! {
+            <div class=("bp3-progress-bar", self.props.intent, Some("bp3-no-animation").filter(|_| !self.props.animate.clone()), Some("bp3-no-stripes").filter(|_| !self.props.stripes.clone()))>

```

Comment:
```suggestion
            <div
                class=(
                    "bp3-progress-bar",
                    self.props.intent,
                    Some("bp3-no-animation").filter(|_| !self.props.animate),
                    Some("bp3-no-stripes").filter(|_| !self.props.stripes)
                )
            >
```
Split it up

Reply:
If you use `ConditionalClass` you can make it more readable. The feature will land in Rust stable at some point:

```suggestion
            <div
                class=(
                    "bp3-progress-bar",
                    self.props.intent,
                    self.props.no_animation.map_some("bp3-no-animation"),
                    self.props.no_stripes.map_some("bp3-no-stripes"),
                )
            >
```

But the implementation I did won't work on false easily. Maybe try renaming the fields to `no_animation` and `no_stripes`. In any case, I think it's actually better to have booleans to be `false` by default and `true` if toggled by the user as this pattern is more common.

Reply:
I meant to stay close to the original Blueprint API, are there any guidelines in this area?

Reply:
We should stay "closish" I would say. https://discord.gg/NAFcwhp

I think we need to improve on what they did. In any case we won't be able to reproduce it exactly at 100%. My idea is that we should get the features but use our own implementation of what makes sense in Rust and is comfortable to use with Rust.

Also because JS is a dynamic language while Rust isn't so we will have differences, there is no way we can avoid that.